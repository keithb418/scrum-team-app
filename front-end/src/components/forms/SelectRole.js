import React from "react";
import { FormGroup, ControlLabel, FormControl } from "react-bootstrap";
import PropTypes from "prop-types";

const SelectRole = ({ roles = [], selected = "", onSelect }) => {
  return (
    <FormGroup controlId="roleSelect">
      <ControlLabel>Role</ControlLabel>
      <FormControl
        name="role"
        defaultValue={selected}
        componentClass="select"
        placeholder="Select Role"
        onChange={ onSelect }>

        <option>Select Role</option>

        {roles.map(role =>
          <option
            value={role.name}
            key={role._id}>{role.name}</option>
        )}

      </FormControl>
    </FormGroup>
  );
};

SelectRole.propTypes = {
  onSelect: PropTypes.func,
  roles: PropTypes.array,
};

export default SelectRole;