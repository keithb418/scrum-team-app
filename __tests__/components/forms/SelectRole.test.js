import React from "react";
import { shallow } from "enzyme";
import SelectRole from "../../../front-end/src/components/forms/SelectRole";
import rolesData from '../../fixtures/roles';

let onSelect, wrapper
test('should render the SelectRole component correctly', () => {
    onSelect = jest.fn();
    wrapper = shallow(
      <SelectRole 
          roles={rolesData} 
          onSelect={onSelect}/>
    );
  expect(wrapper).toMatchSnapshot();
});
