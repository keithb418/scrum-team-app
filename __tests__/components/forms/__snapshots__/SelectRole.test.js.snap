// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the SelectRole component correctly 1`] = `
<FormGroup
  bsClass="form-group"
  controlId="roleSelect"
>
  <ControlLabel
    bsClass="control-label"
    srOnly={false}
  >
    Role
  </ControlLabel>
  <FormControl
    bsClass="form-control"
    componentClass="select"
    defaultValue=""
    name="role"
    onChange={[MockFunction]}
    placeholder="Select Role"
  >
    <option>
      Select Role
    </option>
    <option
      key="1"
      value="Front-End Developer"
    >
      Front-End Developer
    </option>
    <option
      key="2"
      value="Back-End Developer"
    >
      Back-End Developer
    </option>
    <option
      key="3"
      value="Full-Stack Developer"
    >
      Full-Stack Developer
    </option>
    <option
      key="4"
      value="Manual Tester"
    >
      Manual Tester
    </option>
    <option
      key="5"
      value="Automation Tester"
    >
      Automation Tester
    </option>
  </FormControl>
</FormGroup>
`;
